contracts/templates/ERC20Template3.sol

```diff
@@ -0,0 +1,1184 @@
+    address private paymentCollector;
+    address private publishMarketFeeAddress;
+    address private publishMarketFeeToken;
+    uint256 private publishMarketFeeAmount;
+    uint256 private rate = 0;
+    uint256 public constant BASE = 1e18;
+
+    // -------------------------- PREDICTOOR --------------------------
+    enum Status {
+        Pending,
+        Paying,
+        Canceled
+    }
+    event PredictionSubmitted(
+        address indexed predictoor,
+        uint256 indexed slot,
+        uint256 stake
+    );
+    event PredictionPayout(
+        address indexed predictoor,
+        uint256 indexed slot,
+        uint256 stake,
+        uint256 payout,
+        bool prediction,
+        bool truval,
+        Status status
+    );
+    event NewSubscription(
+        address indexed user,
+        uint256 expires,
+        uint256 blocknum
+    );
+    event TruevalSubmitted(
+        uint256 indexed slot,
+        bool trueval,
+        Status status
+    );
+    struct Prediction {
+        bool predval;
+        uint256 stake;
+        address predictoor;
+        bool paid;
+    }
+    struct Subscription {
+        address user;
+        uint256 expires;
+    }
+
+    event SettingChanged(
+        uint256 blocks_per_epoch,
+        uint256 blocks_per_subscription,
+        uint256 truval_submit_timeout_block,
+        address stakeToken
+    );    
+
+    // All mappings below are using slot as key.  
+    // Whenever we have functions that take block as argumens, we rail it to slot automaticly
+    mapping(uint256 => mapping(address => Prediction)) private predobjs; // id to prediction object
+    mapping(uint256 => uint256) private agg_predvals_numer;
+    mapping(uint256 => uint256) private agg_predvals_denom;
+    mapping(uint256 => bool) public truevals;
+    mapping(uint256 => Status) public truval_submitted;
+    mapping(uint256 => uint256) private subscription_revenue_at_block; //income registred
+    mapping(address => Subscription) public subscriptions; // valid subscription per user
+    uint256 public blocks_per_epoch;
+    address public stake_token;
+    uint256 public blocks_per_subscription;
+    uint256 public truval_submit_timeout_block;
+    bool public paused = false;
+    // -------------------------- PREDICTOOR --------------------------
+
+    // EIP 2612 SUPPORT
+    bytes32 public DOMAIN_SEPARATOR;
+    // keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
+    bytes32 public constant PERMIT_TYPEHASH =
+        0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9;
+
+    mapping(address => uint256) public nonces;
+    address public router;
+
+    struct fixedRate {
+        address contractAddress;
+        bytes32 id;
+    }
+    fixedRate[] fixedRateExchanges;
+    address[] dispensers;
+
+    struct providerFee {
+        address providerFeeAddress;
+        address providerFeeToken; // address of the token
```

Comment:
Change comment to: `// token which provider fees are denominated in`

Why: clarity

Reply:
see above